pygac/pod_reader.py

```diff
@@ -443,13 +444,14 @@ def _adjust_clock_drift(self):
             line_indices = (self.scans["scan_line_number"]
                             + int_offsets)
 
-            missed = sorted((set(line_indices) |
-                             set(line_indices + 1))
-                            - set(self.scans["scan_line_number"]))
+            missed = sorted(
+                (set(range(line_indices.min(), self.scans["scan_line_number"].max() + 1))
+                 | set(line_indices + 1))
+                - set(self.scans["scan_line_number"])
+            )
 
             min_idx = min(line_indices)
-            max_idx = max(max(line_indices),
-                          max(self.scans["scan_line_number"] - min_idx)) + 1
+            max_idx = max(line_indices.max(), self.scans["scan_line_number"].max()) + 1
```

Comment:
why remove `min_idx` here?

Reply:
because this leads to the index error. The `max` argument compares line numbers and is therefore in the line number domain. the `- min_idx` is the transformation to the `complete_lons` array index domain. In the case of the crashing orbits, we had exactly the situation, where the right side wins the max comparison.

Reply:
Maybe better variable names could help to avoid this confusion. Something like `min_line, max_line`, `line_range = max_line - min_line`. Then we can use index or `idx` for the indices of `complete_lons, complete_lats` which are numbers in `0:line_range+1`, where the interesting ones are `self.scans["scan_line_number"] - min_line`.

Reply:
> because this leads to the index error. The `max` argument compares line numbers and is therefore in the line number domain. the `- min_idx` is the transformation to the `complete_lons` array index domain. In the case of the crashing orbits, we had exactly the situation, where the right side wins the max comparison.

From reading again the code, I see, that the right hand side should win the max comparison whenever the clock drift is positive, because the array `line_indices` results from 
https://github.com/pytroll/pygac/blob/5c59f2be9437ad7bfc0005038f45f8416d10828d/pygac/pod_reader.py#L438-L444
where offsets is a floating point number in seconds. The multiplication with `-2` to convert the time offset into a line number offset is not well documented, but we know that the sign of the offset determines who wins the max comparison. I think the reason why it sill worked in most of the cases is the `+2` in the following line:
https://github.com/pytroll/pygac/blob/5c59f2be9437ad7bfc0005038f45f8416d10828d/pygac/pod_reader.py#L453
the line I tested that crashed had an `int_offset` of `-3`, which in combination with the unexplained `+2` appeared suspicious to me. Since the max index has already the `+1`, the range of valid indices should not require any `+2`. My first guess was that the `+2` is required for the later interpolation, but slerp is linear and therefore only requires the next neighbours.

Reply:
Now, I assume that it also requires this change to cover all possible indexes
```
min_idx = min(line_indices.min(), self.scans["scan_line_number"].min())
```

Reply:
And without understanding the transformation, but arguing with symmetry, I would assume that 
https://github.com/pytroll/pygac/blob/5c59f2be9437ad7bfc0005038f45f8416d10828d/pygac/pod_reader.py#L440
should be changed to
```python
int_offsets = np.where(offsets<0, np.floor(offsets), np.ceil(offsets)).astype(np.int) 
```
to cover the right index range, but here I am not yet 100% sure.